review/fileToReview.js

```diff
@@ -1,73 +1,67 @@
- * @param {Number} id Account id
+ * @param {Number} accountId Account id
  * @param {Number} page Number of the page
- * @param {Object} previousTransactions Previous page of transactions (To ckeck for dupes)
- * @return {Object} All transactions available on the page
+ * @param {Object} previousTransactions Previous page of transactions (To check for dupes)
+ * @return {Object} allTransactions transactions available on the page
  */
-async function fetchTransactions(fromDate, authorization, jws = null, id, page, previousTransactions) {
-	console.log(`--- Fetch Trasactions page n°${page} ---`);
-	try {
-    var headers = {"Authorisation":  authorization }
+const fetchTransactions = async (
+  fromDate,
+  authorization,
+  jws = null,
+  accountId,
+  page,
+  previousTransactions) => {
+  console.log(`--- Fetch Transactions page n°${page} ---`);
+  try {
+    let headers = {
+      "Authorisation":  authorization,
+      "Content-type": "application/json",
+      "Accept": "application/json"
+    };
 
     if (jws) {
-      headers = {
-        "Authorisation": authorization,
-        "jws": jws,
-        "Content-type": "application/json",
-        "Accept": "application/json"
-      }
-    } else {
-      headers = {
-        "Authorisation": authorization,
-        "Content-type": "application/json",
-        "Accept": "application/json",
-      }
+      headers.jws = jws;
     }
 
-	  var {code, response } = await doRequest('GET',
-      domain + '/accounts/'+ id + '/transactions?' + `page=${page}`,
+    const {code, response} = await doRequest('GET',
+      `${domain}/accounts/${accountId}/transactions?page=${page}`,
       headers);
 
 
-		if (response && code == 200 && response.data) {
-      if (response.data.meta) {
-        if (response.data.meta.hasPageSuivante) {
-          let mouvements = response.data.Mouvements;
-          var date = mouvements[mouvements.length -1].dateValeur;
-          if (date <= fromDate) {
-            console.log("FromDate is Reached - we don't need more transaction");
-          } else {
-            // if we have mouvements
-            if (mouvements) {
-              if (assertTransactions(mouvements)) {
-                return [];
-              } else {
-                console.log(`Push transactions from page ${page}`);
-              }
+    if (code === 200 && response.data && response.data.meta && response.data.meta.hasNextPage) {
+      if (response.data.Mouvements) {
+        let mouvements = response.data.Mouvements;
+        const date = mouvements[mouvements.length - 1].valueDate;
+        if (date <= fromDate) {
+          console.log("FromDate is Reached - we don't need more transaction");
+        } else {
+          if (mouvements) {
+            if (assertTransactions(mouvements)) {
+              return [];
             } else {
-              throw new Error("Empty list of transactions ! " + JSON.stringify(previousTransactions));
```

Comment:
oubli dans le commit: si on décide d'utiliser un format pour stringifier, il faut garder le même partout pour la cohérence => dans ce cas il faudrait: `Empty list of transactions ! ${JSON.stringify(previousTransactions)}`